### PR TITLE
feat: Introduce the compose package

### PR DESCRIPTION
**Pull request by LucaSeri**
### Prerequisite checklist

  - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
  - [x] Tested your changes against relevant architectures and platforms;
  - [x] Ran `make fmt` on your commit series before opening this PR;
  - [ ] Updated relevant documentation.

### Description of changes

The compose package provides primitives for handling complex ``Unikraft`` aplication via the ``Compose`` specification. At the moment, the package only contains basic functionalities for validating a project and packaging all the services to prepare them for running.

### CODE REVIEW

File: compose/project.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+// NewProjectFromComposeFile loads a compose file and returns a project. If no
+// compose file is specified, it will look for one in the current directory.
+func NewProjectFromComposeFile(ctx context.Context, composeFile string) (*Project, error) {
+	candidateFiles := []string{
+		"docker-compose.yml",
+		"docker-compose.yaml",
+		"compose.yml",
+		"compose.yaml",
+	}
```
Comment (nderjung) [inline compose/project.go]: Let's expose these as public variable, `DefaultFileNames` and add `Composefile` to the mix :)

File: compose/project.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+	if composeFile == "" {
+		log.G(ctx).Debugf("No compose file specified, looking for one in the current directory")
```
Comment (nderjung) [inline compose/project.go]: This debug line can be removed

File: compose/project.go [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/compose-spec/compose-go/loader"
+	"github.com/compose-spec/compose-go/types"
+
+	"kraftkit.sh/log"
+	mplatform "kraftkit.sh/machine/platform"
+
```
Comment (nderjung) [inline compose/project.go]: Extra space can be removed

**Review by nderjung — COMMENTED** [review]
Just a couple of nit-picks, other than that, looks good!

**Comment by LucaSeri** [comment]
Thanks @nderjung, I added your suggested changes.

**Review by nderjung — APPROVED** [review]
Awesome! Thanks!

**Pull request merged**